src/components/app/MenuButtons/MetaInfo.js

```diff
@@ -270,27 +268,8 @@ function _formatDate(timestamp: number): string {
   return timestampDate;
 }
 
-function _formatVersionNumber(version?: string): string | null {
-  const regex = /[0-9]+.+[0-9]/gi;
-
-  if (version) {
-    const match = version.match(regex);
-    if (match) {
-      return match.toString();
-    }
-  }
-  return null;
-}
-
-function _formatLabel(meta: ProfileMeta): string {
-  const product = meta.product || '';
-  const version = _formatVersionNumber(meta.misc) || '';
-  let os;
-  // To displaying Android Version instead of Linux for Android developers.
-  if (meta.platform !== undefined && meta.platform.match(/android/i)) {
-    os = meta.platform;
-  } else {
-    os = meta.oscpu || '';
-  }
-  return product + (version ? ` (${version})` : '') + (os ? ` ${os}` : '');
+function _formatMetainfoString(meta: ProfileMeta) {
+  const productAndVersion = formatProductAndVersion(meta);
+  const os = formatPlatform(meta);
+  return productAndVersion + (os ? ` – ${os}` : '');
 }
```

Comment:
I kept it here because in the future it will disappear: indeed we'll display the 2 parts in 2 different elements, because we'll also want to add small icons to represent the platforms and products. Very probably this will all go to a different component.